src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -240,7 +241,18 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 		// while the UI is not yet ready. Clients have to deal with this fact and we have to make sure that the
 		// stacks model gets updated if any of the UI updating fails with an error.
 		const group = this.ensureGroup(position, !options || !options.preserveFocus);
-		const pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		let pinned = !this.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+		if (this.previewEditors) {
+			pinned = !this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>().workbench.editor.enablePreviewFromQuickOpen;
```

Comment:
It is fine to push down services like that but if you have a more elegant solution, also fine.

So you are talking about introducing a new global setting to control previewing from quick open? I honestly thing we do not need that because in other quick opens we already behave like that (e.g. quick outline, goto line, change theme). We can just have it on by default 👍